telemetry/telemetry.go

```diff
@@ -143,44 +145,43 @@ func getDistinctID() string {
 	}
 
 	// Try to derive a distinct ID from the credentials, if present, even
-	// if the /v1/user call failed.
+	// if the getUser() call failed.
 	keyID := cfg.DistinctIDFromCredentials()
 	if keyID != "" {
-		return keyID
+		return keyID, ""
```

Comment:
Maybe use "unknown" if we are sending this string as-is?  I don't see a check for "" in the use site, and it feels like dealing with the empty string would be difficult in Amplitude?